Mathlib/Algebra/MvPolynomial/Degrees.lean

```diff
@@ -261,6 +261,24 @@ theorem degreeOf_mul_le (i : σ) (f g : MvPolynomial σ R) :
   convert Multiset.count_le_of_le i (degrees_mul f g)
   rw [Multiset.count_add]
 
+theorem degreeOf_pow_le (i : σ) (p : MvPolynomial σ R) (n : ℕ) :
+    degreeOf i (p ^ n) ≤ n * degreeOf i p := by
```

Comment:
Can you prove the equality under `NoZeroDivisors`?

Reply:
Yes absolutely! Sorry for being unclear

Reply:
Hmm, I think this is not immediate. The way I proved `degreeOf_pow_le`, it depends on `degrees_pow` which depends on `degrees_prod` which depends on `supDegree_prod_le`. And I don't know there is a version `supDegree_prod_eq` with equality.

So it seems like I'd need a different route to prove the equality version, which will both non-trivial and will complicate this PR. Also, most of the theorems in this file only states the `le` version and not `eq`.

Reply:
Could you add a TODO then?

Reply:
Done, thanks!

Reply:
Oh, looks like you have added `NoZeroDivisors` to the `le` version, probably because you were experimenting with the `eq` version. Can you revert?

Reply:
Yes, sorry! I think the `NoZeroDivisors` condition should be enough for proving equality for arbitrary products, right?

Sketch of argument: let `p = X_i^a * p_a(X not i) + ...` and `q = X_i^b * q_b(X not i) + ...`. Then `degreeOf i (p * q) = a + b` unless `p_a * q_b = 0`, which should not happen since `NoZeroDivisors R` implies `NoZeroDivisors (MvPolynomial σ R)`, which we can prove by induction.